### PR TITLE
009 add dummy parser

### PR DESCRIPTION
**Pull request by retkiewi**

### CODE REVIEW

File: converter.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    Z  0.0  30.0   {mesh_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(mesh_nx=self.x, mesh_ny=self.y, mesh_nz=self.z)
+
+
+class BeamConfig:
```
Comment (grzanka) [inline converter.py:54]: use dataclass as well

File: converter.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    def __init__(self, energy: float = 250., nstat: int = 10000) -> None:
+        self.energy = energy
+        self.nstat = nstat
+
+    def __str__(self) -> str:
+        return self.beam_template.format(energy=self.energy, nstat=self.nstat)
+
+
+class MatConfig:
```
Comment (grzanka) [inline converter.py:75]: use dataclass as well

File: converter.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    def __init__(self, materials: list[int] = [276]) -> None:
+        self.materials: list[str] = materials
+
+    def __str__(self) -> str:
+        material_strings = [self.material_template.format(
+            idx=idx, mat=mat) for idx, mat in enumerate(self.materials)]
+        return "\n".join(material_strings)
+
+
+class DetectConfig:
```
Comment (grzanka) [inline converter.py:92]: use dataclass as well

File: converter.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    def __init__(self, geometries: list[Geometry] = [Cylinder(), Mesh()]) -> None:
+        self.geometries: Geometry = geometries
+
+    def __str__(self):
+        detect_strings = [str(geo)
+                          for geo in self.geometries] + [self.detect_template]
+        return "\n".join(detect_strings)
+
+
+class GeoConfig:
```
Comment (grzanka) [inline converter.py:109]: use dataclass as well

File: converter.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    def __init__(self) -> None:
+        self.beam_config = BeamConfig()
+        self.mat_config = MatConfig()
+        self.detect_config = DetectConfig()
+        self.geo_config = GeoConfig()
+
+    def parse_configs(self, json: dict):
+        pass
+
+    def print_configs(self, target_dir: str):
```
Comment (grzanka) [inline converter.py:160]:
```suggestion
    def save_configs(self, target_dir: str):
```

**Comment by grzanka** [comment]
Add tests as well

File: converter.py (hunk 4 above)
Comment (retkiewi) [reply converter.py:111]: Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f

File: converter.py (hunk 3 above)
Comment (retkiewi) [reply converter.py:93]: Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f

File: converter.py (hunk 1 above)
Comment (retkiewi) [reply converter.py:55]: Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f

File: converter.py (hunk 2 above)
Comment (retkiewi) [reply converter.py:76]: Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f

File: converter.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    material_template: str = """MEDIUM {idx}
+ICRU {mat}
+END
+"""
+
+    def __init__(self, materials: list[int] = [276]):
+        self.materials: list[str] = materials
+
+    def __str__(self) -> str:
+        material_strings = [self.material_template.format(
```
[the remark this answers is not in the document]
Comment (retkiewi) [reply converter.py:87]: Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f

File: converter.py (hunk 5 above)
Comment (retkiewi) [reply converter.py]: Resolved in 3745118fadacd3eefeaf2cee5b591052a549786f

File: editor_with_comments.json.commented [hunk 7]
```diff
@@ -0,0 +1,552 @@
+{
```
[the remark this answers is not in the document]
Comment (retkiewi) [reply editor_with_comments.json.commented:1]: Moved to input_files in 3745118fadacd3eefeaf2cee5b591052a549786f

[the remark this answers is not in the document]
Comment (retkiewi) [reply editor_with_comments.json.commented:1]: Input files renamed to input_examples in 3a1b790d22b0edd0bbde1178ab0508e0bec1507e

**Comment by retkiewi** [comment]
Tests in progress

**Comment by grzanka** [comment]
@retkiewi 
Clean directory structure, now examples are doubled:
https://github.com/yaptide/converter/tree/009-Add-dummy-parser

image

**Comment by grzanka** [comment]
Before adding tests:
- [ ] move all *.py to a directory called `parser` or `converter`
- [ ] create `__init__.py` file in that directory
- [ ] create separate directory for tests: `tests`
- [ ] add some dummy github actions config which runs pytest (for inspiration https://github.com/yaptide/yaptide/blob/master/.github/workflows/test-run.yml)

File: converter/converter.py [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+    Quantity Dose
+    """
+
+    def __str__(self):
+        detect_strings = [str(geo)
+                          for geo in self.geometries] + [self.detect_template]
+        return "\n".join(detect_strings)
+
+
+@ dataclass
```
Comment (grzanka) [inline converter/converter.py:115]:
```suggestion
@dataclass
```

File: converter/converter.py [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+"""
+
+    def __str__(self) -> str:
+        return self.geo_template
+
+
+class Parser(ABC):
+    """Abstract parser, the template for implementing other parsers."""
+
+    @ abstractmethod
```
Comment (grzanka) [inline converter/converter.py:146]:
```suggestion
    @abstractmethod
```

File: converter/converter.py [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+    def parse_configs(json: dict):
+        """Convert the json dict to the 4 config dataclasses."""
+        pass
+
+    @ abstractmethod
```
Comment (grzanka) [inline converter/converter.py:151]:
```suggestion
    @abstractmethod
```

File: converter/converter.py [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+
+class Runner:
+    """Converts input data dict to files that will be saved in output_dir using the specified parser."""
+
+    def __init__(self, parser: Parser, input_data: dict, output_dir: str) -> None:
+        self.parser = parser
+        self.input_data = input_data
+        self.output_dir = output_dir
+
+    def run_parser(self):
```
Comment (grzanka) [inline converter/converter.py:199]: add type hinting for return type

Comment (retkiewi) [reply converter/converter.py]: Resolved in b951def677fac830fd2757dc6fb68fb4a8c5e5e5

File: converter/converter.py (hunk 10 above)
Comment (retkiewi) [reply converter/converter.py]: Resolved in b951def677fac830fd2757dc6fb68fb4a8c5e5e5

File: converter/converter.py (hunk 9 above)
Comment (retkiewi) [reply converter/converter.py]: Resolved in b951def677fac830fd2757dc6fb68fb4a8c5e5e5

File: converter/converter.py (hunk 8 above)
Comment (retkiewi) [reply converter/converter.py]: Resolved in b951def677fac830fd2757dc6fb68fb4a8c5e5e5

**Review by grzanka — APPROVED** [review]
Looks ok

**Pull request merged**